contracts/libraries/Pool.sol

```diff
@@ -58,22 +64,21 @@ library Pool {
     /// @notice Thrown by donate if there is currently 0 liquidity, since the fees will not go to any liquidity providers
     error NoLiquidityToReceiveFees();
 
-    /// The uint8 fees variables are represented as integer denominators (1/x)
-    /// For swap fees, the upper 4 bits are the fee for trading 1 for 0, and the lower 4 are for 0 for 1 and are taken as a percentage of the lp swap fee.
-    /// For withdraw fees the upper 4 bits are the fee on amount1, and the lower 4 are for amount0 and are taken as a percentage of the principle amount of the underlying position.
-    /// swapFee:     1->0  | 0->1
-    /// withdrawFee: fee1  | fee0
+    /// Each uint24 variable packs both the swap fees and the withdraw fees. The upper 12 bits are the swap fees, and the lower 12 bits are the withdraw fees.
+    /// For swap fees, the upper 6 bits are the fee for trading 1 for 0, and the lower 6 are for 0 for 1 and are taken as a percentage of the lp swap fee.
+    /// For withdraw fees the upper 6 bits are the fee on amount1, and the lower 6 are for amount0 and are taken as a percentage of the principle amount of the underlying position.
+    ///                   withdrawFees    |    swapFees
```

Comment:
```suggestion
    /// bits          0  2  4  6  8  10 12 14 16 18 20 22 24
    ///               |  withdrawFees   |    swapFees     |
    ///               ┌────────┬────────┬────────┬────────┐
    /// protocolFees: | fee1   |  fee0  |  1->0  |  0->1  |
    /// hookFees:     | fee1   |  fee0  |  1->0  |  0->1  |
    ///               └────────┴────────┴────────┴────────┘
```

Reply:
1-> 0 | 0->1 should be for all swap fees, not protocol fees

Reply:
Ok I see. I think all that needed correction was the comment since this was how the code was already working. However, I think the comment should be updated to
```
    /// bits          0  2  4  6  8  10 12 14 16 18 20 22 24
    ///               |  withdrawFees   |    swapFees     |
    ///               ┌────────┬────────┬────────┬────────┐
    /// protocolFees: | fee1   |  fee0  |  0->1  |  1->0  |
    /// hookFees:     | fee1   |  fee0  |  0->1  |  1->0  |
    ///               └────────┴────────┴────────┴────────┘
```

The update here is from from `1-> 0 | 0->1` to 0-> 1 | 1->0

Pool.sol L418
```
        SwapCache memory cache = SwapCache({
            liquidityStart: self.liquidity,
            protocolFee: params.zeroForOne
                ? (getSwapFee(slot0Start.protocolFees) % 64)
                : (getSwapFee(slot0Start.protocolFees) >> 6),
            hookFee: params.zeroForOne ? (getSwapFee(slot0Start.hookFees) % 64) : (getSwapFee(slot0Start.hookFees) >> 6)
        });
```

From this code, (if `zeroForOne` represents `0 -> 1`), then from the way the code works, it seems the upper bits are used for `1 -> 0`.

Reply:
Yah feel free to make sure the comment matches the code obviously. so the right bits of the swap are `0->1` and the left bits of swap are `1-> 0`. It also looks like swapFees are on the left and withdraw fees are on the right, so that'd be a good change to make too!!!

Reply:
So it depends on how 'left' and 'right' are defined. I prefer to use the terms 'upper' and 'lower'. In the way the picture is drawn, upper bits are on the right and upper bits are indeed used for swap fees, so that is already correct.

Pool.sol L136
```
function getSwapFee(uint24 feesStorage) internal pure returns (uint16) {
    return uint16(feesStorage >> 12);
}
```

However, `fee1` and `fee0` do need to be flipped. So this will be the final correct comment:

Reply:
I see what you're saying. We use "shift right" a lot in the fee logic, so I think the concept of right and left is pretty baked in. What if we switched the bit index's so lsb is on the right `24 <-- 0` ??

Reply:
Yeah so this would also work.

```
/// bits          24 22 20 18 16 14 12 10 8  6  4  2  0
///               |    swapFees     |   withdrawFees  |
///               ┌────────┬────────┬────────┬────────┐
/// protocolFees: | 1->0   |  0->1  |  fee1  |  fee0  |
/// hookFees:     | 1->0   |  0->1  |  fee1  |  fee0  |
///               └────────┴────────┴────────┴────────┘
```

Would you prefer it this way?

Reply:
yes --^ this looks good to me

Reply:
This version looks good to me too^ !